lib/zkProving.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  MembershipProver,
+  Tree,
+  Poseidon,
+  defaultPubkeyMembershipPConfig,
+} from "@personaelabs/spartan-ecdsa";
+import { loadSigmojis, updateSigmoji } from "@/lib/localStorage";
+import { cardPubKeys } from "./cardPubKeys";
+import { importPublic, ecrecover } from "@ethereumjs/util";
+import { sha256 } from "js-sha256";
+import { serializeSigmojiZKP } from "@/lib/types";
+import { parseDERSignature, findV } from "@/lib/signatureUtils";
+
+export interface ProverWasm {
+  poseidon: Poseidon;
+  prover: MembershipProver;
+}
+
+export async function initWasm(): Promise<ProverWasm> {
+  const poseidon = new Poseidon();
+  await poseidon.initWasm();
+
+  const prover = new MembershipProver({
+    ...defaultPubkeyMembershipPConfig,
+    enableProfiler: true,
+  });
+
+  await prover.initWasm();
+
+  return { poseidon, prover };
+}
+
+export const makeProofs = async (proverWasm: ProverWasm) => {
+  const treeDepth = 20;
```

Comment:
Do we need a tree this large?

Reply:
We don't, but that's just what the current spartan-ecdsa circuits provide for. We could recompile their circuits with a lower tree size (we only need depth 5) but I'm not sure if it's worth going through that effort given we are hoping to not use this proof past Berlin.